ouroboros-consensus/src/Ouroboros/Consensus/Demo/Run.hs

```diff
@@ -1,117 +1,51 @@
-  deriving (Show)
-
-{-------------------------------------------------------------------------------
-  Constraints required to run the demo
--------------------------------------------------------------------------------}
-
-class DemoHeaderHash hh where
-  demoEncodeHeaderHash :: hh -> Encoding
-  demoDecodeHeaderHash :: Decoder s hh
-
-class ( ProtocolLedgerView blk
-      , DemoHeaderHash (HeaderHash blk)
-      , Condense (Header blk)
-      , Condense (ChainHash blk)
-      , Condense (HeaderHash blk)
-      , Condense blk
-      , Condense [blk]
-      -- TODO: the show constraints are needed when using a tracer which logs
-      -- a mini-protocol messages, or the fetch client decisions, we should
-      -- rethink where the condense class lives so that we can provide
-      -- 'Condenese' instances for data structures defined in ouroboros-network
-      -- or provide them in the module where 'Condense' is defined.
-      , Show blk
-      , Show (Header blk)
-      , ApplyTx blk
-      , Show (ApplyTxErr blk)
-      , Condense (GenTx blk)
-      , Show (GenTx blk)
-      , Show (Header blk)
-      ) => RunDemo blk where
-  demoForgeBlock         :: (HasNodeState (BlockProtocol blk) m, MonadRandom m)
-                         => NodeConfig (BlockProtocol blk)
-                         -> SlotNo         -- ^ Current slot
-                         -> BlockNo        -- ^ Current block number
-                         -> ChainHash blk  -- ^ Previous hash
-                         -> [GenTx blk]    -- ^ Txs to add in the block
-                         -> IsLeader (BlockProtocol blk)
-                         -> m blk
-  demoBlockMatchesHeader :: Header blk -> blk -> Bool
-  demoBlockFetchSize     :: Header blk -> SizeInBytes
-  demoIsEBB              :: blk -> Bool
-  demoEpochSize          :: Monad m
-                         => Proxy blk  -> EpochNo -> m EpochSize
-
-  -- Encoders
-  demoEncodeBlock       :: NodeConfig (BlockProtocol blk) -> blk                            -> Encoding
-  demoEncodeHeader      :: NodeConfig (BlockProtocol blk) -> Header blk                     -> Encoding
-  demoEncodeGenTx       ::                                   GenTx  blk                     -> Encoding
-  demoEncodeLedgerState :: NodeConfig (BlockProtocol blk) -> LedgerState blk                -> Encoding
-  demoEncodeChainState  :: Proxy blk                      -> ChainState (BlockProtocol blk) -> Encoding
-
-  -- Decoders
-  demoDecodeHeader      :: forall s. NodeConfig (BlockProtocol blk) -> Decoder s (Header blk)
-  demoDecodeBlock       :: forall s. NodeConfig (BlockProtocol blk) -> Decoder s blk
-  demoDecodeGenTx       :: forall s.                                   Decoder s (GenTx blk)
-  demoDecodeLedgerState :: forall s. NodeConfig (BlockProtocol blk) -> Decoder s (LedgerState blk)
-  demoDecodeChainState  :: forall s. Proxy blk                      -> Decoder s (ChainState (BlockProtocol blk))
-
+class RunNode blk => RunDemo blk where
```

Comment:
ok I see this is just to run the tests suite / demo.  So forget my previous comment.